### DISCUSSION TITLE
Working with a knowledge base

### DISCUSSION THREAD

**Discussion by deema-A** [facebookresearch/ParlAI#4805] [Q&A]
Hi, 

I am new to ParlAI, I want to build a QA chatbot that gets the gist of the question and then sends it to the wizard, and then the chatbot propagates the wizard's answer to the user. 

Example:
user: what is the best drink in the world?
the chatbot sends (the best drink in the world), so the wizard suggests it is Coffee.
bot: I believe coffee is the best drink, I love it.

Not sure how and where to start building this. if you can kindly suggest examples of ParlAI repositories that print/show the user's main question at each step I will appreciate it a lot.

Best,

**Comment by mojtaba-komeili** [comment]
There is the natural question dataset that is very close to this. We also have Wizard of Internet dataset and FITS that use search, if you think your data needs to come from an external source. I would start with those as the starting dataset. And maybe fine-tune SKR agents on them.
